### PR TITLE
Addresses #210

### PR DESCRIPTION
**Pull request by Tam-Lin**
Moves the os.environ declaration to a patch, to prevent unintended (or intended) side-effects.

### CODE REVIEW

File: chaoslib/secret.py [hunk 1]
```diff
@@ -74,74 +74,60 @@ def load_secrets(secrets_info: Dict[str, Dict[str, str]],
...
     extra_vars = extra_vars or {}
-    for loader in loaders:
-        for key, value in loader(
-                secrets_info, configuration, extra_vars).items():
-            if key not in secrets:
-                secrets[key] = {}
-            secrets[key].update(value)
 
-    logger.debug("Secrets loaded")
+    logger.debug("extra_vars: %s" % extra_vars)
```
Comment (Lawouach) [inline chaoslib/secret.py:81]: Mmmh, isn't this a tad risk to expose in the logs?

File: chaoslib/secret.py [hunk 2]
```diff
@@ -74,74 +74,60 @@ def load_secrets(secrets_info: Dict[str, Dict[str, str]],
...
+            logger.debug("Need to further process %s" % value)
 
-def load_inline_secrets(secrets_info: Dict[str, Dict[str, str]],
-                        configuration: Configuration = None,
-                        extra_vars: Dict[str, Any] = None) -> Secrets:
-    """
-    Load secrets that are inlined in the experiments.
-    """
-    secrets = {}
+            logger.debug("extra_vars[key]: %s" % extra_vars.get(key, None))
```
Comment (Lawouach) [inline chaoslib/secret.py:89]: I'm not sure exposing secrets in the logs is a good idea?

File: chaoslib/secret.py [hunk 3]
```diff
@@ -74,74 +74,60 @@ def load_secrets(secrets_info: Dict[str, Dict[str, str]],
...
-            elif value.get("type") not in ("env", "vault"):
-                secrets[target][key] = extra_vars.get(target, {}).get(
-                    key, value)
+            if extra_vars.get(key, None) is not None:
+                secrets[key] = extra_vars.get(key)
 
-        if not secrets[target]:
-            secrets.pop(target)
+            elif value.get("type") is "env":
+                logger.debug("env processing")
```
Comment (Lawouach) [inline chaoslib/secret.py:95]: Noisy?

File: chaoslib/secret.py [hunk 4]
```diff
@@ -74,74 +74,60 @@ def load_secrets(secrets_info: Dict[str, Dict[str, str]],
...
+                secrets[key] = extra_vars or load_secret_from_env(value)
+
+            elif value.get("type") is "vault":
+                logger.debug("vault processing")
```
Comment (Lawouach) [inline chaoslib/secret.py:99]: likewise, a bit noisy

File: chaoslib/secret.py [hunk 5]
```diff
@@ -74,74 +74,60 @@ def load_secrets(secrets_info: Dict[str, Dict[str, str]],
...
+                secrets[key] = load_secret_from_env(value, configuration, extra_vars)
+
+            else:
+                secrets[key] = load_secrets(value, configuration, extra_vars.get(key, None))
+
+        else:
+            secrets[key] = value
+
+    logger.debug(secrets)
```
Comment (Lawouach) [inline chaoslib/secret.py:108]: probably not secure

File: chaoslib/secret.py [hunk 6]
```diff
@@ -74,74 +74,60 @@ def load_secrets(secrets_info: Dict[str, Dict[str, str]],
...
 
+def load_secret_from_env(secrets_info: Dict[str, Dict[str, str]]) -> Secrets:
 
-def load_secrets_from_env(secrets_info: Dict[str, Dict[str, str]],
-                          configuration: Configuration = None,
-                          extra_vars: Dict[str, Any] = None) -> Secrets:
-    env = os.environ
-    secrets = {}
+    logger.debug("load_secret_from_env")
+    logger.debug("secrets_info: %s" % secrets_info)
```
Comment (Lawouach) [inline chaoslib/secret.py:116]: probably not required anymore?

**Review by Lawouach — CHANGES_REQUESTED** [review]
Thanks for the first draft, woth cleaning up the debug statements.

Outputing secrets in log files isn't a good idea I think.

Make sure to run pycodestyle to delint.

Could you also add a changelog entry explaining the change and linking to the issue?

**Comment by Tam-Lin** [comment]
Apologies; not all of this should have been made into a push request yet.  The change for nested processing is still being worked on; all this pull request was supposed to be was moving the os.environ into a patch.  If you want to reject it, feel free, and I'll resubmit once it's ready.

**Comment by Lawouach** [comment]
Gotcha. Happy to have a dedicated PR for the patching of os.environ specifically (with a changelog entry please). I'll reject this one so you can submit two PRs (-one for the patch and one for the reworking on secrets laoding).

Cheers

**Pull request closed without merging**